### ISSUE TITLE
Resource Problem with keras.utils.multi_gpu_model

### ISSUE THREAD

**Issue by Raziel1511** [keras-team/keras#8425]
# Resource Problem with keras.utils.multi_gpu_model
Hello,

I'm using Keras v.2.0.9 to train an Autoencoder. It worked fine on a single GPU. To increase the training speed I used `keras.utils.multi_gpu_model` to train on 8 GPUs parallel:

```python
# Build the network
input_data = Input(shape=(self.cuboid_dim[0], self.cuboid_dim[1], self.cuboid_dim[2], 1, ))

reduced_data = self.encode(input_data) # Method similar to reconst_data
reconst_data  = self.decode(reduced_data) # Method explained below

# Model for Autoencoder
with tf.device('/cpu:0'):
        autoencoder = Model(input_data, reconst_data)
        parallel_autoencoder = multi_gpu_model(autoencoder, gpus=8)

# Compile the Autoencoder
sgd = optimizers.SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
parallel_autoencoder.compile(optimizer=sgd, loss='mean_squared_error')

# Get the data 
data = self.getData()

# Train the Autoencoder
parallel_autoencoder.fit(data, data, epochs=1000, batch_size=128)

def decode(self, x):
     flatten_layer_dec = Flatten(input_shape=(8, 8, 8, 1))(x)
     flatten_layer_dec = BatchNormalization()(flatten_layer_dec)  
     reconst_layer_1 = Dense(64000, activation='sigmoid')(flatten_layer_dec) 
     reconst_layer_2 = Dense(1000, activation='sigmoid')(reconst_layer_1)
     return Reshape(target_shape=(10, 10, 10, 1))(reconst_layer_2)

```

On a single GPU my model worked fine, but on multiple GPUs (8 GTX1080) I was facing two issues:

- I had to decrease the size of the Dense Layers, otherwise I ran oom (e.g. the Dense layer was two magnitudes bigger)
- After roughly 20 iteration the program crashes with the error: `OOM when allocating tensor with shape[64000,1000]`  with the information:

`[[Node: training/SGD/gradients/replica_0/model_1/Reconst_Layer_2/MatMul_grad/MatMul_1 = MatMul[T=DT_FLOAT, _class=["loc:@replica_0/model_1/Reconst_Layer_2/MatMul"], transpose_a=true, transpose_b=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](replica_0/model_1/Reconst_Layer_1/Sigmoid, training/SGD/gradients/replica_0/model_1/Reconst_Layer_2/Sigmoid_grad/SigmoidGrad)]]`

Am I doing something wrong here or is it a bug?

---
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by pstalidis** [comment]
I think you are instantiating you parallel_autoencoder in the cpu.
In the example:
```
# Instantiate the base model
# (here, we do it on CPU, which is optional).
with tf.device('/cpu:0'):
    model = Xception(weights=None,
                     input_shape=(height, width, 3),
                     classes=num_classes)

# Replicates the model on 8 GPUs.
# This assumes that your machine has 8 available GPUs.
parallel_model = multi_gpu_model(model, gpus=8)
parallel_model.compile(loss='categorical_crossentropy',
                       optimizer='rmsprop')
```
the parallel model is instantiated outside the cpu

**Comment by Raziel1511** [comment]
It's the very same in the example given by Keras. I think this forces Keras to synchronize shared variables on the CPU, that are updated by the GPUs after each Batch/ Epoch.
